trin-execution/src/execution.rs

```diff
@@ -255,11 +319,32 @@ impl State {
                     Transaction::AccessList(tx) => tx.modify(block_number, tx_env),
                     Transaction::Blob(tx) => tx.modify(block_number, tx_env),
                 }
-            })
-            .with_external_context(NoOpInspector)
-            .append_handler_register(inspector_handle_register)
-            .build()
-            .transact()?)
+            });
+
+        Ok(
```

Comment:
I think something like this is much nicer:
```rust
let mut evm_builder = Evm::builder()
    ...; // part from above

if self.config.block_to_trace.should_trace(block_number) {
  ...
  evm_builder = evm_builder.with_external_context(
      TracerEip3155::new(Box::new(output_file)));
} else {
  evm_builder = evm_builder.with_external_context(NoOpInspector);
}

Ok(evm_builder
    .append_handler_register(inspector_handle_register)
    .build()
    .transact()?)
```

Reply:
I tried earlier, but I get this 
<img width="577" alt="image" src="https://github.com/ethereum/trin/assets/31669092/e54ba954-7cdd-4106-aa39-358ceb3cdf75">

Reply:
I tried your example quite a few things and I am not sure this example or idea is possible. I originally tried your example when I first tried implementing this a few days ago, but I couldn't get it to compile